src/nvim/lua/executor.c

```diff
@@ -1406,6 +1406,7 @@ char_u *nlua_register_table_as_callable(typval_T *const arg)
   lua_State *const lstate = nlua_enter();
 
   int top = lua_gettop(lstate);
```

Comment:
What about changing this to

```suggestion
#ifndef NDEBUG
  int top = lua_gettop(lstate);
#endif
```